i hope i did it right

qwerq12311 opened the pull request:
i hope i did it right

qwerq12311 committed:
+ main + usercreation tests
- need to figure out with deletion

qwerq12311 committed:
+ main
+ 1. user creation tests + figured out with deletion

+ 2. Login

qwerq12311 committed:
1. almost done. need to rebuild OrderCreationTests's logic.
2. Need to create reports.

qwerq12311 committed:
3.all done

must do - reports

qwerq12311 committed:
3.all done

3.1 must do - reports - done

mkouznetsov-git commented on src/main/java/ru/yandex/diplom_2/ApiClient.java:
```diff
@@ -0,0 +1,236 @@
+    public static final String BASE_URL = "https://stellarburgers.nomoreparties.site/api";
+    private static final String ENDPOINT_INGREDIENTS = "/ingredients";
+    private static final String ENDPOINT_CREATE_ORDER = "/orders";
+    private static final String ENDPOINT_RESET_PASSWORD = "/password-reset";
```
Эта константа нигде не используется

qwerq12311 replied:
1. Исправлено : удалено.

mkouznetsov-git commented on src/main/java/ru/yandex/diplom_2/ApiClient.java:
```diff
@@ -0,0 +1,236 @@
+    private static final String ENDPOINT_INGREDIENTS = "/ingredients";
+    private static final String ENDPOINT_CREATE_ORDER = "/orders";
+    private static final String ENDPOINT_RESET_PASSWORD = "/password-reset";
+    private static final String ENDPOINT_RESET_PASSWORD_RESET = "/password-reset/reset";
```
Эта константа нигде не используется

qwerq12311 replied:
1. Исправлено : удалено.

mkouznetsov-git commented on src/main/java/ru/yandex/diplom_2/ApiClient.java:
```diff
@@ -0,0 +1,236 @@
+    // Методы для создания заказа
+
+    public static Response createOrderWithRandomIngredientsWithoutToken() {
+        List<String> ingredientIds = getIngredientIds();
```
Переменная лишняя, она нигде не используется

qwerq12311 replied:
1.Исправлено : удалено.

mkouznetsov-git commented on src/main/java/ru/yandex/diplom_2/ApiClient.java:
```diff
@@ -0,0 +1,236 @@
+    }
+
+
+    public static Response createOrder(String[] ingredients, String accessToken) {
```
Входящий параметр ingredients не используется в методе, удаляй его, если он не нужен.

mkouznetsov-git commented on src/main/java/ru/yandex/diplom_2/ApiClient.java:
```diff
@@ -0,0 +1,236 @@
+    }
+
+    public static Response createOrderWithRandomIngredients(String accessToken) {
+        List<String> ingredientIds = getIngredientIds();
```
Переменная ingredientIds не используется

qwerq12311 replied:
1.Исправлено : удалено.

mkouznetsov-git commented on src/main/java/ru/yandex/diplom_2/ApiClient.java:
```diff
@@ -0,0 +1,236 @@
+        requestBody.addProperty("password", userPassword);
+        requestBody.addProperty("name", name);
+
+        Response response = post(ENDPOINT_REGISTER_USER, requestBody.toString()); // Передаем JSON-строку
```
Лишняя переменная, можно сразу return...

mkouznetsov-git commented on src/test/java/ru/yandex/diplom_2/UserCreationTest.java:
```diff
@@ -0,0 +1,107 @@
+import io.restassured.response.Response;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import static io.restassured.RestAssured.*;
```
Импорт не используется

qwerq12311 replied:
1. Исправлено : удалено.

mkouznetsov-git commented on src/test/java/ru/yandex/diplom_2/UserCreationTest.java:
```diff
@@ -0,0 +1,107 @@
+    @Before
+    public void setup() {
+        ApiClient.setup(); // Вызываем метод из ApiClient для установки базового URL
+        apiClient = new ApiClient();
```
Нет смысла инстанцировать утилитный класс (когда все его методы статичные)

qwerq12311 replied:
1. Исправлено : удалено.

mkouznetsov-git commented on src/test/java/ru/yandex/diplom_2/UserCreationTest.java:
```diff
@@ -0,0 +1,107 @@
+    @After
+    public void tearDown() {
+        if (createdUserAccessToken != null) {
+            apiClient.deleteUser(createdUserAccessToken);
```
К статичным методам нужно обращаться по имени класса, а не через объект. В остальных местах нужно это тоже поправить.

qwerq12311 replied:
1. Исправлено : к статичным методам обращение происходит по имени класса.

mkouznetsov-git commented on src/test/java/ru/yandex/diplom_2/UserLoginTest.java:
```diff
@@ -0,0 +1,88 @@
+import io.restassured.response.Response;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import static io.restassured.RestAssured.*;
```
Импорт не используется

qwerq12311 replied:
1. Исправлено : удалено.

mkouznetsov-git commented on src/test/java/ru/yandex/diplom_2/UserLoginTest.java:
```diff
@@ -0,0 +1,88 @@
+    @Before
+    public void setup() {
+        ApiClient.setup();
+        apiClient = new ApiClient();
```
Статика.

qwerq12311 replied:
1. Исправлено : удалено.

qwerq12311 committed:
1. Удалил все неиспользуемые импорты и переменные
2. Переработал логику метода создания заказа, обновил тесты
3. Исправил использование статичных методов

qwerq12311 committed:
1. Удалил неиспользуемый метод
